pkg/apis/core/validation/shoot.go

```diff
@@ -2887,90 +2915,125 @@ func ValidateCoreDNSRewritingCommonSuffixes(commonSuffixes []string, fldPath *fi
 	return allErrs
 }
 
-func validateShootOperation(operation, maintenanceOperation string, shoot *core.Shoot, fldPath *field.Path) field.ErrorList {
+func validateShootOperation(operations, maintenanceOperations []string, shoot *core.Shoot, fldPath *field.Path) field.ErrorList {
 	var (
+		fldPathOp          = fldPath.Key(v1beta1constants.GardenerOperation)
+		fldPathMaintOp     = fldPath.Key(v1beta1constants.GardenerMaintenanceOperation)
 		allErrs            = field.ErrorList{}
 		encryptedResources = sets.New[schema.GroupResource]()
 		k8sLess134, _      = versionutils.CheckVersionMeetsConstraint(shoot.Spec.Kubernetes.Version, "< 1.34")
 	)
 
-	if operation == "" && maintenanceOperation == "" {
+	if len(operations) == 0 && len(maintenanceOperations) == 0 {
+		return allErrs
+	}
+
+	if len(strings.Join(operations, ";")) > maxOperationsSize {
+		allErrs = append(allErrs, field.TooLong(fldPathOp, len(operations), maxOperationsSize))
```

Comment:
Would it be possible instead of handling and checking the max size of the operations annotation to deduplicate and clean earlier in the queue?
After all, iiuc, there are only so many operations possible at a given moment at all (at the moment max. ~16 + possibly longer values, e.g. for explicit rotations).

How about in `PrepareForUpdate` (`pkg/apiserver/registry/core/shoot/strategy.go:258`), we already add them to a set while doing the `SplitAndTrimString` handling?

That way, we ensure that for the subsequent validations, the operations are deduplicated and clean.

WDYT? Also @plkokanov 😃